ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/SnapServer.java

```diff
@@ -100,7 +100,10 @@ class SnapServer implements BesuEvents.InitialSyncCompletionListener {
       final EthMessages snapMessages,
       final WorldStateStorageCoordinator worldStateStorageCoordinator,
       final ProtocolContext protocolContext) {
-    this.snapServerEnabled = snapConfig.isSnapServerEnabled();
+    this.snapServerEnabled =
+        Optional.ofNullable(snapConfig)
```

Comment:
Why Snapconfig can be null? In general there is a default configuration?

Reply:
Several tests mock SyncConfiguration and have a null snap config.  We can change the 5 tests where this mocking is happening, or address the potential NPE here.  Would you prefer that we `orElse(SnapSyncConfiguration.getDefault())` ?